webhook.go

```diff
@@ -0,0 +1,20 @@
+package plugin
+
+import "github.com/gin-gonic/gin"
+
+// WebhookRegisterRequest consists of the custom router mux and the auth middlewares that can be used plugins.
+type WebhookRegisterRequest struct {
+	Mux         gin.RouterGroup
+	MiddleWares struct {
+		RequireApplicationAuth gin.HandlerFunc
+		RequirePluginToken     gin.HandlerFunc
```

Comment:
Do we need this, I thought the url would already contain the token: https://github.com/gotify/server/issues/51#issuecomment-451739352
> `/plugin/:name/custom/:randUserToken/*` - the * path is available for the plugin to configure

where randUserToken would be the plugin token.

Reply:
I am not sure about this either:( This would be used if some plugins need to expose hooks that are not private to the user itself(eg: an `administrator announcement` plugin could register a handler that can be only called by an admin and forward the message to the user)

Reply:
This should be some kind of global plugin I guess, because here you do not have a list of users and also you can't create applications for users. I would say we ignore something like this and only support plugins for one user.

Reply:
Okay, but I would still preserve the struct as maybe more information would need to be provided while registering handlers in the future

Reply:
Does this make a difference? Changing the struct will still kill compatibility right?

Reply:
Yeah, you are right we can remove this.